app/counters/workflow_counter.rb

```diff
@@ -30,4 +30,56 @@ def retired_subjects
   def launch_date
     workflow.project.launch_date
   end
+
+  # this is an optimized query to access the subject_id index on this table.
+  # It uses an IN query with a CTE to create a subselect to access
+  # the subject_id index as simple Joins do not use the index on this table
+  #
+  # WITH sws_by_set AS (
+  #   SELECT set_member_subjects.subject_id
+  #   FROM set_member_subjects
+  #   WHERE set_member_subjects.subject_set_id = 1
+  # )
+  # SELECT SUM(classifications_count)
+  # FROM subject_workflow_counts
+  # WHERE subject_workflow_counts.subject_id
+  # IN (SELECT subject_id FROM sws_by_set)
+  def sws_query(select_clause)
+    cte = sws_by_set_cte(workflow.subject_sets.pluck(:id))
+    select_manager = sws_by_set_select
+    query = SubjectWorkflowStatus.where(select_manager)
+
+    query = yield(query)
+
+    query = query.select(select_clause)
```

Comment:
Wondering if it would be better to delegate this to the block as well?

Reply:
Good point, i should have done this as well.